src/main/java/ru/vk/itmo/test/vadimershov/ShardingDao.java

```diff
@@ -0,0 +1,133 @@
+package ru.vk.itmo.test.vadimershov;
+
+import one.nio.http.HttpException;
+import one.nio.http.Response;
+import one.nio.pool.PoolException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import ru.vk.itmo.ServiceConfig;
+import ru.vk.itmo.dao.Dao;
+import ru.vk.itmo.dao.Entry;
+import ru.vk.itmo.test.vadimershov.exceptions.DaoException;
+import ru.vk.itmo.test.vadimershov.exceptions.NotFoundException;
+import ru.vk.itmo.test.vadimershov.exceptions.RemoteServiceException;
+import ru.vk.itmo.test.vadimershov.hash.ConsistentHashing;
+import ru.vk.itmo.test.vadimershov.hash.VirtualNode;
+
+import java.io.IOException;
+import java.lang.foreign.MemorySegment;
+
+import static ru.vk.itmo.test.vadimershov.utils.MemorySegmentUtil.toByteArray;
+import static ru.vk.itmo.test.vadimershov.utils.MemorySegmentUtil.toDeletedEntity;
+import static ru.vk.itmo.test.vadimershov.utils.MemorySegmentUtil.toEntity;
+import static ru.vk.itmo.test.vadimershov.utils.MemorySegmentUtil.toMemorySegment;
+
+public class ShardingDao {
+
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+    public static final String ENTITY_URI = "/v0/entity?id=";
+
+    private final String selfUrl;
+    private final Dao<MemorySegment, Entry<MemorySegment>> localDao;
+    private final ConsistentHashing consistentHashing;
+
+    public ShardingDao(ServiceConfig serviceConfig, Dao<MemorySegment, Entry<MemorySegment>> localDao) {
+        this.selfUrl = serviceConfig.selfUrl();
+        this.localDao = localDao;
+        this.consistentHashing = new ConsistentHashing(serviceConfig.clusterUrls());
+    }
+
+    public byte[] get(String key) {
+        VirtualNode virtualNode = consistentHashing.findVNode(key);
+        if (virtualNode.url().equals(selfUrl)) {
+            Entry<MemorySegment> entry;
+            try {
+                entry = localDao.get(toMemorySegment(key));
+            } catch (Exception e) {
+                logger.error(e.getMessage());
```

Comment:
Принято так (и во всех остальных местах) -- что мы пытались сделать, входные параметры, ошибка:
```suggestion
                logger.error("Can't get value by key={}", key, e);
```